### DISCUSSION TITLE
p:menu jumps and moves without any action

### DISCUSSION THREAD

**Discussion by kadirbekmn** [primefaces/community#3142] [PrimeFaces] [answered]
Hi there. I am using p:menu for my website. With Primefaces14 i realized that menu doesn't work properly. Before PF14, i didn't have any problem with menu but now it makes me angry, when i scrool down on menu.

The interesting part is, i have tried this movement on primefaces official showcase page and i happens also on their website. Probably they didn't yet realize this problem. 

Note: Menu have to scroll for this problem. I mean, u can zoom in to have scrool on menu. And then you can also realize this problem.
Note: I use toggleable menu.

So, if u guys have any solution, pls share with me :)!

![image](https://github.com/user-attachments/assets/2f314b96-e896-43ac-bab6-f8daffb05b0c)

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by kadirbekmn** [reply]
Actually, I was about to provide an executable example but which is interesting, this problem happens also at Primefaces Official Website. That's why, i thought that i don't need it. 

https://www.primefaces.org/showcase/ui/menu/menu.xhtml?jfwid=f7d25

**Reply by kadirbekmn** [reply]
https://github.com/user-attachments/assets/239ce7ed-cf76-4651-a737-517813d50647

**Reply by kadirbekmn** [reply]
There are cuts in video to make it shorter but at the seconds that i wrote "here happens" there is not cut. Menu jumps by itself.

**Reply by melloware** [reply]
@kadirbekmn its because the Menus are now ARIA compliant and accessible.  see ticket: https://github.com/primefaces/primefaces/issues/6561

what is happening is the highlighted menu gets `tabIndex="0"` so the screen reader can read it for visually impaired users but I believe the browser is now scrolling it into view if its off screen.  Prior to 14.0.1 our menus were not screen reader compliant which many Government agencies require.

**Reply by kadirbekmn** [reply]
I got it. Thanks to inform me. So, what's gonna happen now? Should I find a solution by myself to block this movement? Or are you gonna work for this menu movement to fix it? According your answer, I will start to work so that's why I need to know :) Btw, thanks for your support and help.

**Reply by melloware** [reply]
well if you can figure out some way to not make it happen we can maybe make a fix but I don't want to break ARIA Accessibility as that is more important.

**Reply by kadirbekmn** [reply]
Well, I will start to work on it so do you have any advice to start? For example is there any possible way to close the ARIA Accessibility on my local? Or a tip that can help me to solve this problem?

**Reply by melloware** [reply, marked as answer]
I didn't test this but if you apply this MonkeyPatch by adding this JS to some JS file that loads in your app it will override the default behavior and I "think" turn off what you want..

```js
if (PrimeFaces.widget.PlainMenu) {
  PrimeFaces.widget.PlainMenu.prototype.bindOverlay = function() {
    var $this = this;

    if (this.cfg.overlay) {
      // Hide menu when any menu item link is clicked
      this.menuitemLinks.on("click", function() {
        $this.hide();
      });

      // Handle keyboard navigation for overlay
      this.trigger.on('keydown.ui-menu', function(e) {
        switch (e.key) {
          case 'ArrowDown':
            if (!$this.jq.is(':visible')) {
              $this.show();
            }
            e.preventDefault();
            break;

          case 'Tab':
            if ($this.jq.is(':visible')) {
              $this.hide();
            }
            break;
        }
      });
    }
  };
};
```
